include/aspect/material_model/rheology/composite_visco_plastic.h

```diff
@@ -94,30 +112,36 @@ namespace aspect
           compute_strain_rate_and_derivative (const double creep_stress,
                                               const double pressure,
                                               const double temperature,
+                                              const unsigned int composition,
                                               const DiffusionCreepParameters diffusion_creep_parameters,
                                               const DislocationCreepParameters dislocation_creep_parameters,
                                               const PeierlsCreepParameters peierls_creep_parameters,
-                                              const double min_viscosity,
-                                              const double max_viscosity) const;
+                                              const DruckerPragerParameters drucker_prager_parameters) const;
 
         private:
 
           /**
-           * Whether to use different creep mechanisms
+           * Whether to use different deformation mechanisms
            */
           bool use_diffusion_creep;
           bool use_dislocation_creep;
           bool use_peierls_creep;
+          bool use_drucker_prager;
 
           /**
-           * Pointers to objects for computing viscous creep viscosities.
+           * Pointers to objects for computing deformation mechanism
+           * strain rates and effective viscosities.
            */
           std::unique_ptr<Rheology::DiffusionCreep<dim>> diffusion_creep;
           std::unique_ptr<Rheology::DislocationCreep<dim>> dislocation_creep;
           std::unique_ptr<Rheology::PeierlsCreep<dim>> peierls_creep;
+          std::unique_ptr<Rheology::DruckerPrager<dim>> drucker_prager;
+
+          DruckerPragerParameters drucker_prager_parameters;
 
-          std::vector<double> min_viscosities;
-          std::vector<double> max_viscosities;
+          unsigned int number_of_compositions;
+          double min_viscosity;
+          double max_viscosity;
```

Comment:
Can you remind me the reason for switching to single min/max viscosities? I recall we discussed this, but cannot remember the details. In any event, I agree it makes sense to have a single min/max viscosity that applied to all deformation mechanisms.

Reply:
I think the argument was that as it's just a stabilising parameter, we shouldn't give the user freedom for no good reason. 
Although I was originally ambivalent about the choice, single min/max parameters will also make for tidier formulations when we implement different averaging schemes.

Reply:
👍